include/deal.II/base/tensor_product_polynomials.h

```diff
@@ -79,6 +79,8 @@ class TensorProductPolynomials
   template <class Pol>
   TensorProductPolynomials (const std::vector<Pol> &pols);
 
+  virtual ~TensorProductPolynomials() {};
+
```

Comment:
I originally needed that for the dynamic_cast. With types_are_equal there is no longer a need for this class to be virtual.